### DISCUSSION TITLE
Enable linking libraries in deployContract()

### DISCUSSION THREAD

**Discussion by pistomat** [wevm/viem#2564] [Idea / Feature Request]
Viem currently does not support linking libraries on deployment. Compiled ABI include `linkReferences` list of placeholders in bytecode that should be replaced with an address of a deployed library. Hardhat with Ethers has this feature already:
https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-ethers#library-linking

It would be great if Viem also had this support. The logic should be pretty straightforward, replace the placeholder listed in `linkReferences` in a for loop with supplied addresses.

More context:
https://docs.soliditylang.org/en/latest/using-the-compiler.html#library-linking

**Comment by clemsos** [comment]
is that supported today?

**Reply by clemsos** [reply]
answering my own question: yes, it is. https://github.com/NomicFoundation/hardhat/pull/4695/files

here is an example

```js
  const auroraXccUtils = await hre.viem.deployContract('AuroraXccUtils')
  const codec = await hre.viem.deployContract('Codec')
  const auroraSdk = await hre.viem.deployContract('AuroraSdk', [], {
    libraries: {
      AuroraXccUtils: auroraXccUtils.address,
      Codec: codec.address,
    },
  })
```
